### PR TITLE
Small updates to TargetCoverageSexGenotyper

### PR DESCRIPTION
**Pull request by mbabadi**
See https://github.com/broadinstitute/gatk-protected/issues/715

### CODE REVIEW

**Comment by mbabadi** [comment]
@asmirnov239, this is a tiny pull request -- can you take a look at your earliest convenience?

File: src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelUtils.java [hunk 1]
```diff
@@ -14,63 +14,48 @@
...
      *
-     *     n_t \sim \mathrm{Poisson}(\rho l_t c_t m_t p_t)
+     *     n_t \sim \mathrm{Poisson}(d c_t m_t p_t)
      *
-     * where $n_t$ is the read count, $l_t$ is target length, $c_t$ is copy ratio, $p_t$ is ploidy, and $m_t$ is
+     * where $n_t$ is the read count, $c_t$ is copy ratio, $p_t$ is ploidy, and $m_t$ is
      * the multiplicative bias (the estimation method is described in CNV-methods.pdf).
      *
-     * Note: the mask array is 0 or 1, and targets with 0 mask will be effectively dropped out.
+     * Note: the mask array {@param mask} 0 or 1, and targets with 0 mask will be effectively dropped out.
```
Comment (asmirnov239) [inline src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelUtils.java]: "the mask array {@param mask} consists of 0s or 1s, and targets with..."

File: src/test/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelUtilsUnitTest.java [hunk 2]
```diff
@@ -16,67 +16,60 @@
...
+                                                                final int[] ploidies,
+                                                                final double[] copyRatio,
+                                                                final int[] mask) {
+        CoverageModelUtils.estimateReadDepthFromPoissonModel(readCounts, multiplicativeBiases,
                 ploidies, copyRatio, mask);
     }
 
-    @DataProvider(name = "estimateReadDepthDensityExceptionsDataProver")
-    public Object[][] estimateReadDepthDensityExceptionsDataProver() {
+    @DataProvider(name = "estimateReadDepthExceptionsDataProver")
```
Comment (asmirnov239) [inline src/test/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelUtilsUnitTest.java]: should this be `estimateReadDepthExceptionsDataProvider`?

File: src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/SexGenotypeDataCollection.java [hunk 3]
```diff
@@ -44,13 +46,19 @@ public SexGenotypeDataCollection(@Nonnull final File sexGenotypeDataFile) throws
      * @param sexGenotypeDataReader an instance of {@link Reader}
```
Comment (asmirnov239) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/SexGenotypeDataCollection.java]: these are out of order

File: src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/SexGenotypeDataCollection.java [hunk 4]
```diff
@@ -60,6 +68,8 @@ public SexGenotypeDataCollection(@Nonnull final String sexGenotypeDataSourceName
      */
     public void add(@Nonnull final SexGenotypeData sexGenotypeData) {
         sexGenotypeDataList.add(sexGenotypeData);
+        sampleNames.add(sexGenotypeData.getSampleName());
+        sampleNamesToSexGenotypeDataMap.put(sexGenotypeData.getSampleName(), sexGenotypeData);
```
Comment (asmirnov239) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/SexGenotypeDataCollection.java]: Do you want to also check here that the new sample name is unique?

File: src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/TargetCoverageSexGenotypeCalculator.java [hunk 5]
```diff
@@ -169,13 +173,34 @@ public SexGenotypeDataCollection inferSexGenotypes() {
...
-                copyRatio, mask);
+        final double[] readCounts = processedReadCounts.getColumnOnSpecifiedTargets(sampleIndex,
+                autosomalTargetList, false);
+        final double[] readCountsNormalizedByPloidy = IntStream.range(0, readCounts.length)
+                .mapToDouble(i -> readCounts[i] / (double)autosomalTargetPloidies[i])
+                .toArray();
+        return new Median().evaluate(readCountsNormalizedByPloidy);
+
+        /**
+         * TODO
```
Comment (asmirnov239) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/TargetCoverageSexGenotypeCalculator.java]: can you just leave a note here saying what needs to be done to fix it?

File: src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelUtils.java (hunk 1 above)
Comment (mbabadi) [inline src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelUtils.java]: done

File: src/test/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelUtilsUnitTest.java (hunk 2 above)
Comment (mbabadi) [inline src/test/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelUtilsUnitTest.java]: done

File: src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/SexGenotypeDataCollection.java (hunk 3 above)
Comment (mbabadi) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/SexGenotypeDataCollection.java]: done

File: src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/SexGenotypeDataCollection.java (hunk 4 above)
Comment (mbabadi) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/SexGenotypeDataCollection.java]: done, good catch!

File: src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/TargetCoverageSexGenotypeCalculator.java (hunk 5 above)
Comment (mbabadi) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/TargetCoverageSexGenotypeCalculator.java]: done, good idea.

**Pull request merged**